src/sensitivities/linalg/generic.jl

```diff
@@ -1,7 +1,7 @@
 # Implementation of sensitivities for unary linalg optimisations.
 unary_linalg_optimisations = [
     (:-,          ∇RealArray, ∇RealArray, :(map(-, Ȳ)),                        (lb, ub)),
-    (:trace,      ∇RealArray, ∇Real,      :(Diagonal(ones(size(X, 1)))),       (lb, ub)),
+    (:trace,      ∇RealArray, ∇Real,      :(Diagonal(Ȳ .* ones(size(X, 1)))),  (lb, ub)),
```

Comment:
Why `:(Ȳ * Diagonal(ones(size(X, 1))))` (`Ȳ` should be a scalar) instead of `:(Ȳ * eye(size(X, 1)))`? Does using `Diagonal` yields performance gains? I'd love to write `:(Ȳ * I)` here, but that seems to somehow break `compute_Dv`.